jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -24,9 +24,58 @@
 from jdaviz.core.tools import ICON_DIR
 from jdaviz.core.custom_traitlets import IntHandleEmpty
 
+from scipy.interpolate import make_interp_spline
+
 __all__ = ['PlotOptions']
 
 
+class SplineStretch:
+    """
+    A class to represent spline stretches.
+
+    Attributes:
+        k (int): Degree of the smoothing spline.
+        bc_type (str or None): Boundary condition type.
+        t (array-like or None): Array of knot positions.
+        x (array-like): The x-coordinates of the data points.
+        y (array-like): The y-coordinates of the data points.
+        spline (object): Interpolating spline.
+    """
+
+    def __init__(self):
+        """
+        Initializes the SplineStretch class with default knot values.
+        """
+        self.k = 3
+        self.bc_type = None
+        self.t = None
+        self.update_knots(
+            x=np.array([0, 0.1, 0.2, 0.7, 1]),
+            y=np.array([0, 0.05, 0.3, 0.9, 1])
+        )
+
+    def __call__(self, values, out=None, clip=False):
+        # TODO: Review the 'out' and 'clip' parameters once glue upstream is updated.
+        # Depending on their implementation, this method may need adjustment.
+        return self.spline(values)
+
+    def update_knots(self, x, y):
+        if len(x) != len(y):
+            raise ValueError("x and y must be the same length.")
+        self.x = x
+        self.y = y
+        self._update_spline()
+
+    def _update_spline(self):
+        self.spline = make_interp_spline(
+            self.x, self.y, k=self.k, t=self.t, bc_type=self.bc_type
+        )
+
+
+# Add the spline stretch to the glue stretch registry
+stretches.add("spline", SplineStretch(), display="Spline")

```

Comment:
What does this line do if someone executes this Python file multiple times? Does it add duplicate entries into `stretches`? Do we need to prevent that?

Reply:
I think this raises an error, so if we can check if its in the registry first before adding, that would be nice to have!

Reply:
Thank you both for your input. As @kecnry  mentioned, it throws an error and does not allow for re-registration. I'll make sure to update the code as needed.